custom_components/unifiprotect/views.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from http import HTTPStatus
+import logging
+from typing import Any
+
+from aiohttp import web
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.core import HomeAssistant, HomeAssistantError
+from pyunifiprotect.exceptions import NvrError
+
+from .utils import get_ufp_instance_from_entity_id
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ThumbnailProxyView(HomeAssistantView):
+    """View to proxy event thumbnails from UniFi Protect"""
+
+    requires_auth = False
```

Comment:
Why no auth?

Reply:
The plan is to use this in Push Notification URLs so they cannot require auth. The `thumb_id` is already random 24-character value. Should be really hard to brute force. 

I can add another randomized value if you think that is not good enough.

Reply:
Actually, looking at the event IDs more closely. They do not seem completely randomize. They seem to be incrementing in some way. I definitely think I need to add another value to the URL to increase entropy. Let me look how camera_proxy works.

Reply:
After looking at how camera_proxy works and thinking about, I think I am just going to reimplement the camera_proxy pattern onto the motion binary sensors.

I was also planning on using this view later for a media browser to view events and clips, but I think I will make a subclass of this view (or vice versa) and have one that does not require auth/has access token and one that does.

Reply:
Okay, I added the access token pattern that `camera_proxy` uses. Same general idea, but now the URL also uses all UFP values outside of the access token.